backend/homework/2/src/main/java/org/q1/StudentUtil.java

```diff
@@ -0,0 +1,80 @@
+package org.q1;
+
+import java.util.logging.Logger;
+
+public class StudentUtil {
+    static Logger logger= Logger.getLogger(String.valueOf(StudentUtil.class));
+    public static double []calculateGPA(int []studentIdList,char[][]studentsGrades)
+    {
+
+        int totalStudents=studentIdList.length;
+        double []gpa=new double[totalStudents];
+        if(totalStudents==0||studentsGrades.length==0)
+            return gpa;
+        for(int i=0;i<totalStudents;i++)
```

Comment:
Use better names, try making the code modular by making sure you follow SOLID principles. Whereever possible add modified iterative methods like stream but atleast try using for each for better readability